consul/consul.go

```diff
@@ -137,15 +134,15 @@ func (c *Consul) Register(service *registry.Service) {
 //
 func (c *Consul) Deregister() error {
 	for s, b := range serviceCache {
-		if !b.isRegistered {
+		if c.CacheIsValid(s) {
+			c.CacheProcessDeregister(s)
+		} else {
```

Comment:
If it is left at the default of 1 it will actually behave similarly to how mesos-consul behaved before this change. If mesos-consul fails to find the running task in the state even once it will remove it from the consul registry